File: dotnet/Solid/RefactoredSolid/ISP/Example1/JsonStore.cs
using System;

namespace RefactoredSolid.ISP.Example1
{
    public class JsonStore : IFileConnection, IFileOperation
    {
        public string Filename { get; set; }

        public void Open()
        {
            Console.WriteLine("Opening file");
        }

        public void Close()
        {
            Console.WriteLine("Closing file");
        }

        public string ReadFile()
        {
            return "File content!";
        }
    }
}

File: java/isp-start/src/main/java/no/bouvet/solid/isp/JsonStore.java
package no.bouvet.solid.isp;

public class JsonStore implements PersistenceMedium {

    private String fileName;

    @Override
    public String getFileName() {
        return fileName;
    }

    @Override
    public void setFileName(String fileName) {
        this.fileName = fileName;
    }

    @Override
    public String getConnection() {
        throw new UnsupportedOperationException("Not supported");
    }

    @Override
    public void setConnection(String connection) {
        throw new UnsupportedOperationException("Not supported");
    }

    @Override
    public void open() {
        // open file
    }

    @Override
    public String executeQuery(String query) {
        throw new UnsupportedOperationException("Not supported");
    }

    @Override
    public String readFile() {
        return "File content!";
    }

    @Override
    public void close() {
        // close file
    }

}
